app/session.go

```diff
@@ -34,12 +35,32 @@ func (a *App) CreateSession(session *model.Session) (*model.Session, *model.AppE
 	return session, nil
 }
 
+var UserSessionPool = sync.Pool{
+	New: func() interface{} {
+		return &model.Session{}
+	},
+}
+
 func (a *App) GetSession(token string) (*model.Session, *model.AppError) {
 	metrics := a.Metrics()
 
-	var session *model.Session
+	var session = UserSessionPool.Get().(*model.Session)
+	session.Id = ""
+	session.Token = ""
+	session.CreateAt = 0
+	session.ExpiresAt = 0
+	session.LastActivityAt = 0
+	session.UserId = ""
+	session.DeviceId = ""
+	session.Roles = ""
+	session.IsOAuth = false
+	session.ExpiredNotify = false
+	session.Props = nil
+	session.TeamMembers = nil
+	session.Local = false
```

Comment:
I don't believe there is a need to reset these fields. The unmarshaler should take care of these.

Reply:
thanks, I was not sure exactly how the unmarshaler would override this, removed it as suggested.

Reply:
It will satisfy the Unmarshaler interface, after which it just calls the UnmarshalMsg method which is what we need.